DelvEdit/src/com/interrupt/dungeoneer/editor/ui/menu/generator/LevelGeneratorMenuItem.java

```diff
@@ -84,28 +85,34 @@ private ActionListener makeLevelGeneratorAction(Level template) {
                         if (template != null) {
-                            Level level = Editor.app.getLevel();
-
-                            if (level != null) {
-                                level.editorMarkers.clear();
-                                level.entities.clear();
-                                level.non_collidable_entities.clear();
-                                level.static_entities.clear();
-
-                                level.theme = template.theme;
-                                level.generated = true;
-                                level.dungeonLevel = 0;
-                                level.crop(0, 0, 17 * 5, 17 * 5);
-                                level.roomGeneratorChance = 0.4f;
-                                level.roomGeneratorType = template.roomGeneratorType;
-                                level.generate(Level.Source.EDITOR);
-                                Editor.app.refresh();
-
-                                if (Editor.app.generatorInfo.lastGeneratedLevelTemplate == null
-                                        || Editor.app.generatorInfo.lastGeneratedLevelTemplate.theme != template.theme
-                                        || Editor.app.generatorInfo.lastGeneratedLevelTemplate.roomGeneratorType != template.roomGeneratorType) {
-                                    Editor.app.generatorInfo.lastGeneratedLevelTemplate = template;
-                                    needsRefresh = true;
+                            if (!Editor.app.generatorInfo.isLevelTemplateValid(template)) {
+                                WarningDialog warningDialog = new WarningDialog(skin, "We were not able to find the level template used for this level generator. Make sure it exists.");
+                                warningDialog.show(Editor.app.ui.getStage());
+                            }
+                            else {
+                                Level level = Editor.app.getLevel();
+    
+                                if (level != null) {
+                                    level.editorMarkers.clear();
+                                    level.entities.clear();
+                                    level.non_collidable_entities.clear();
+                                    level.static_entities.clear();
+    
+                                    level.theme = template.theme;
+                                    level.generated = true;
+                                    level.dungeonLevel = 0;
+                                    level.crop(0, 0, 17 * 5, 17 * 5);
+                                    level.roomGeneratorChance = 0.4f;
+                                    level.roomGeneratorType = template.roomGeneratorType;
+                                    level.generate(Level.Source.EDITOR);
+                                    Editor.app.refresh();
+    
+                                    if (Editor.app.generatorInfo.lastGeneratedLevelTemplate == null
```

Comment:
When I see conditionals like this, I typically want to wrap them up in a method to encapsulate/clarify the logic.

Reply:
Fixed.